samples/snippets/src/main/java/com/example/logging/ListLogs.java

```diff
@@ -16,47 +16,34 @@
 
 package com.example.logging;
 
+// [START logging_list_logs]
 import com.google.api.gax.paging.Page;
-import com.google.cloud.logging.LogEntry;
 import com.google.cloud.logging.Logging;
-import com.google.cloud.logging.Logging.EntryListOption;
 import com.google.cloud.logging.LoggingOptions;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.util.Calendar;
 
-/** List logs programmatically using the Cloud Logging API. */
 public class ListLogs {
 
-  /** Expects an existing Cloud Logging log name as an argument. */
   public static void main(String... args) throws Exception {
-    // [START logging_list_log_entries]
-    // Instantiates a client
-    LoggingOptions options = LoggingOptions.getDefaultInstance();
 
-    String logName = args[0];
-
-    try (Logging logging = options.getService()) {
-
-      // When composing a filter, using indexed fields, such as
-      // timestamp, resource.type, logName and others can help accelerate the results
-      // Full list of indexed fields here: https://cloud.google.com/logging/docs/view/advanced-queries#finding-quickly
-      // Below we are restricting the results to only last hour to speedup getting the results back
-      Calendar calendar = Calendar.getInstance();
-      calendar.add(Calendar.HOUR, -1);
-      DateFormat rfc3339 = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-      String logFilter = "logName=projects/" + options.getProjectId() + "/logs/" + logName
-          + " AND timestamp>=\"" + rfc3339.format(calendar.getTime()) + "\"";
-
-      // List all log entries
-      Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(logFilter));
-      do {
-        for (LogEntry logEntry : entries.iterateAll()) {
-          System.out.println(logEntry);
-        }
-        entries = entries.getNextPage();
-      } while (entries != null);
+    LoggingOptions options = LoggingOptions.getDefaultInstance();
+    Logging logging = options.getService();
+
+    // List all log names
+    Page<String> logNames = logging.listLogs();
+    while (logNames != null) {
+      for (String logName : logNames.iterateAll()) {
+        System.out.println(logName);
+      }
+      logNames = logNames.getNextPage();
     }
-    // [END logging_list_log_entries]
   }
 }
+// [END logging_list_logs]
+
+// the following is a temporary location due to exclusive requirement
+// from snippet layout vs. snippet-bot
+
+// [START logging_list_log_entries]
+
+
+// [END logging_list_log_entries]
```

Comment:
If Cloud site is redeployed, the snippet will empty on cloud.google.com which is probably worse than the page failing to deploy.

Reply:
Jeff, it is a problem of the chicken and egg. Your review request was to have one snippet per file (by guidelines). Snippet management team requires the migration to be done by first introducing the code and then moving the region tag (spoke with Takashi). It is impossible to do both at once. This PR is stuck for a week just because we discuss this subject. Please, release it. I will issue the fixing PR within an hour that will clean up the duplicated region tag.
The guidelines allow a short period of time when the particular sample will be empty. I just need the help to take the work forward.